File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        #O(n) - time complexity
    
        l = 0
        countZero = 0
        max_ = 0
        for r in range(len(nums)):
            if nums[r] == 0:
                countZero+=1
            while countZero > k:
                
                if nums[l] == 0:
                    countZero-=1
                l+=1
            max_ = max(max_ , r-l+1)
                
                
                    
                    
               
        return max_
                
        

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int left = 0;
        // int operation = k;
        // int right = 0;
        int longestOne = 0;
  
        int zeroCounter = 0;
        for(int right=0; right<nums.length; right++){
            if(nums[right]==0){
                zeroCounter++;
                
            }
            while(zeroCounter > k){
                if(nums[left] == 0){
                    zeroCounter--;
                    
                }
                left++;
                
                
                
                
            }
            longestOne = Math.max(longestOne , right-left+1);
            
        }
        return longestOne;
        
    }
}